Content.Client/Fluids/PuddleVisualsSystem.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Linq;
+using Content.Shared.Fluids;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Shared.Random;
+
+namespace Content.Client.Fluids
+{
+    [UsedImplicitly]
+    public sealed class PuddleVisualsSystem : VisualizerSystem<PuddleVisualsComponent>
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<PuddleVisualsComponent, ComponentInit>(OnComponentInit);
+        }
+
+        private void OnComponentInit(EntityUid uid, PuddleVisualsComponent puddleVisuals, ComponentInit args)
+        {
+            if (!TryComp(uid, out AppearanceComponent? appearance))
+            {
+                return;
+            }
+
+            if (!TryComp(uid, out SpriteComponent? sprite))
+            {
+                return;
+            }
+
+            if (!appearance.TryGetData(PuddleVisuals.VisualSeed, out float visualSeed))
+            {
+                return;
+            }
+
+            var maxStates = sprite.BaseRSI?.ToArray();
+
+            if (maxStates is not { Length: > 0 }) return;
+
+            int selectedState = (int) Math.Floor(visualSeed * maxStates.Length); // uses the visualSeed to randomly select an index for which RSI state to use.
+            sprite.LayerSetState(PuddleVisualLayers.Puddle, maxStates[selectedState].StateId); // sets the sprite's state via our randomly selected index.
+
+            int rotationDegrees = (int) Math.Floor(visualSeed * 360); // uses the visualSeed to randomly select a rotation for our puddle sprite.
+            sprite.Rotation = Angle.FromDegrees(rotationDegrees); // sets the sprite's rotation to the one we randomly selected.
+        }
+
+
+        protected override void OnAppearanceChange(EntityUid uid, PuddleVisualsComponent component, ref AppearanceChangeEvent args)
+        {
+            if (TryComp(uid, out SpriteComponent? sprite)
+                && args.Component.TryGetData(PuddleVisuals.VolumeScale, out float volumeScale)
+                && args.Component.TryGetData(PuddleVisuals.SolutionColor, out Color solutionColor)
+                && args.Component.TryGetData(PuddleVisuals.WetFloorEffect, out bool wetFloorEffect)
+                )
+            {
```

Comment:
for these large if-blocks, IMO the code is more readable if the logic is inverted. i.e., `if (not condition) { log error; return;}`, rather than having the majority of the function inside of a single if-block, especially when there are nested if-statements

Reply:
Good note! I'm not familiar with tricks like these. I'll correct it here, and keep it in mind going forward as well.